### PR TITLE
Repack diff

### PR DESCRIPTION
**Pull request by kasimi**
So Titania comes with a diff class that makes use of phpBB's core files for diffing that was written in 2008 but was never used. I updated it only where necessary and didn't overhaul it completely.

Notes:
* Previously, each entry in a diff result had a header along the lines of:
    ```
    Index: file.php
    =====================================
    ```
    I removed this.
* The resulting diff always included unchanged files. I added a switch to the diff class to be able to configure it to skip equal files, making the diff smaller and more readable.
* Including phpBB's core files for diffing threw an error about redeclaring classes when using Titania's `_include()` method, which is why I added a new optional argument to skip auto-including. I'm not sure if there's a better way to solve this.
* The `code` BBCode didn't always pick up the `diff`syntax because of too much PHP code in it. This is entirely the fault highlight.js, the library used by s9e/highlighter. I added an even listener that allows using `[code lang=xyz]` throughout Titania which forces the given syntax to be used. Thanks to @JoshyPHP for the code.

Comments and testers are welcome!

repack

### CODE REVIEW

File: language/en/manage.php [hunk 1]
```diff
@@ -126,4 +126,5 @@
 	'VALIDATION_PV'				=> 'Prevalidator Notes',
 	'VALIDATION_QUEUE'			=> 'Validation Queue',
 	'VALIDATION_SUBMISSION'		=> 'Validation Submission',
+    'VALIDATION_DIFF'			=> 'Repack Diff',
```
Comment (DavidIQ) [inline language/en/manage.php]: I think this should probably be `VALIDATION_REPACK_DIFF`.

**Comment by paul999** [comment]
I wonder if we should introduce the same for any revision that is submitted

**Comment by kasimi** [comment]
Those diffs could get quite big though, especially with updated dependencies in the `vendor` folder. Not sure if they are handy enough in all situations.

**Comment by vinny** [comment]
@kasimi @DavidIQ 

Should this only work for extensions? Or for all contributions?It is not working with styles.

**Pull request merged**